### ISSUE TITLE
Error retrieving user karma when user is not authenticated

### ISSUE THREAD

**Issue by fsiddi** [armadillica/dillo#27]
To reproduce, simply click on any post without being logged in.

```
Traceback (most recent call last):
  File "/Users/fsiddi/.venvs/dillo/lib/python3.6/site-packages/eve/flaskapp.py", line 1033, in __call__
    return super(Eve, self).__call__(environ, start_response)
  File "/Users/fsiddi/.venvs/dillo/lib/python3.6/site-packages/flask/app.py", line 1994, in __call__
    return self.wsgi_app(environ, start_response)
  File "/Users/fsiddi/.venvs/dillo/lib/python3.6/site-packages/flask/app.py", line 1985, in wsgi_app
    response = self.handle_exception(e)
  File "/Users/fsiddi/.venvs/dillo/lib/python3.6/site-packages/flask/app.py", line 1540, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/Users/fsiddi/.venvs/dillo/lib/python3.6/site-packages/flask/_compat.py", line 33, in reraise
    raise value
  File "/Users/fsiddi/.venvs/dillo/lib/python3.6/site-packages/flask/app.py", line 1982, in wsgi_app
    response = self.full_dispatch_request()
  File "/Users/fsiddi/.venvs/dillo/lib/python3.6/site-packages/flask/app.py", line 1614, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/Users/fsiddi/.venvs/dillo/lib/python3.6/site-packages/flask/app.py", line 1517, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/Users/fsiddi/.venvs/dillo/lib/python3.6/site-packages/flask/_compat.py", line 33, in reraise
    raise value
  File "/Users/fsiddi/.venvs/dillo/lib/python3.6/site-packages/flask/app.py", line 1612, in full_dispatch_request
    rv = self.dispatch_request()
  File "/Users/fsiddi/.venvs/dillo/lib/python3.6/site-packages/flask/app.py", line 1598, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/Users/fsiddi/Developer/dillo/dillo/posts/routes.py", line 35, in view_embed
    return view_node(node_id=node_id, extra_template_args={'project': project})
  File "/Users/fsiddi/Developer/pillar/pillar/web/nodes/routes.py", line 236, in view
    **extra_template_args)
  File "/Users/fsiddi/.venvs/dillo/lib/python3.6/site-packages/flask/templating.py", line 134, in render_template
    context, ctx.app)
  File "/Users/fsiddi/.venvs/dillo/lib/python3.6/site-packages/flask/templating.py", line 116, in _render
    rv = template.render(context)
  File "/Users/fsiddi/.venvs/dillo/lib/python3.6/site-packages/jinja2/asyncsupport.py", line 76, in render
    return original_render(self, *args, **kwargs)
  File "/Users/fsiddi/.venvs/dillo/lib/python3.6/site-packages/jinja2/environment.py", line 1008, in render
    return self.environment.handle_exception(exc_info, True)
  File "/Users/fsiddi/.venvs/dillo/lib/python3.6/site-packages/jinja2/environment.py", line 780, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/Users/fsiddi/.venvs/dillo/lib/python3.6/site-packages/jinja2/_compat.py", line 37, in reraise
    raise value.with_traceback(tb)
  File "/Users/fsiddi/Developer/dillo/dillo/templates/nodes/custom/dillo_post/view_embed.html", line 38, in top-level template code
    <div class="op-karma">Karma: {{ node.user.extension_props.dillo.karma }}</div>
  File "/Users/fsiddi/.venvs/dillo/lib/python3.6/site-packages/jinja2/environment.py", line 430, in getattr
    return getattr(obj, attribute)
jinja2.exceptions.UndefinedError: 'None' has no attribute 'dillo'
```

**Comment by fsiddi** [comment]
Apparently this is due to the `after_fetching_user` pillar hook, which strips the `extension_props` field. We have to decide whether to allow it, or to handle it in a different way.

**Comment by fsiddi** [comment]
This issue will be fixed soon by creating `extension_props_public` for the user.

**Comment by fsiddi** [comment]
Issue fixed in 07dcf15
Do not forget to run `$ ./manage.py dillo reset_users_karma` in order to recreate the karma prop inside of `extension_props_public`.
